src/main/java/seedu/opus/model/comparators/EndTimeComparator.java

```diff
@@ -0,0 +1,25 @@
+package seedu.opus.model.comparators;
+
+import java.util.Comparator;
+
+import seedu.opus.model.task.ReadOnlyTask;
+
+public class EndTimeComparator implements Comparator<ReadOnlyTask> {
+    public int compare(ReadOnlyTask d1, ReadOnlyTask d2) {
+        boolean hasStartAndEnd = d1.getEndTime().isPresent() && d2.getEndTime().isPresent();
+        boolean hasStartOrEnd = d1.getEndTime().isPresent() || d2.getEndTime().isPresent();
```

Comment:
The variable names confuse me. These two lines check whether both or either task has StartTime or not right? So it should be something like `bothHasStart` or sth? It's the same for StartTimeComparator as well.